src/snyk/snykOss/views/suggestion/suggestionViewProvider.ts

```diff
@@ -0,0 +1,201 @@
+  type: SuggestionViewEventMessageType;
+  value: unknown;
+};
+
+// todo: unify interface between oss and snyk code
+export interface ISuggestionViewProvider {
+  activate(): void;
+  showPanel(vulnerability: OssIssueCommandArg): Promise<void>;
+}
+
+export class SuggestionViewProvider implements ISuggestionViewProvider {
+  private panel?: vscode.WebviewPanel;
+
+  constructor(private readonly context: ExtensionContext, private readonly window: IVSCodeWindow) {}
+
+  activate(): void {
+    this.context.addDisposables(
+      this.window.registerWebviewPanelSerializer(SNYK_VIEW_SUGGESTION_OSS, new SuggestionSerializer(this)),
+    );
+  }
+
+  // todo: perhaps create a base class for both Code and OSS providers?
+  async showPanel(vulnerability: OssIssueCommandArg): Promise<void> {
+    if (
+      !vscode.window.activeTextEditor?.viewColumn ||
+      !this.panel?.viewColumn ||
+      this.panel.viewColumn !== vscode.ViewColumn.Two
+    ) {
+      // workaround for: https://github.com/microsoft/vscode/issues/71608
+      // when resolved, we can set showPanel back to sync execution.
+      await vscode.commands.executeCommand('workbench.action.focusSecondEditorGroup');
+    }
+
+    if (this.panel) {
+      this.panel.reveal(vscode.ViewColumn.Two, true);
+    } else {
+      this.panel = vscode.window.createWebviewPanel(
+        SNYK_VIEW_SUGGESTION_OSS,
+        'Snyk OSS Vulnerability',
+        {
+          viewColumn: vscode.ViewColumn.Two,
+          preserveFocus: true,
+        },
+        {
+          enableScripts: true,
+          localResourceRoots: [this.context.getExtensionUri()],
+        },
+      );
+    }
+
+    this.panel.webview.html = this.getHtmlForWebview(this.panel.webview);
+
+    void this.panel.webview.postMessage({ type: 'set', args: vulnerability });
+
+    this.panel.onDidDispose(this.onPanelDispose.bind(this));
+    this.panel.webview.onDidReceiveMessage((data: FeaturesViewEventMessage) => {
+      switch (data.type) {
+        case SuggestionViewEventMessageType.OpenBrowser:
+          void vscode.commands.executeCommand(SNYK_OPEN_BROWSER_COMMAND, data.value);
+          break;
+        default:
+          break;
+      }
+    });
+    this.panel.onDidChangeViewState(this.checkVisibility.bind(this));
+  }
+
+  restorePanel(panel: vscode.WebviewPanel): void {
+    if (this.panel) this.panel.dispose();
+    this.panel = panel;
+  }
+
+  private disposePanel() {
+    if (this.panel) this.panel.dispose();
+  }
+
+  private onPanelDispose() {
+    this.panel = undefined;
+  }
+
+  private checkVisibility(_e: vscode.WebviewPanelOnDidChangeViewStateEvent): void {
+    if (this.panel && this.panel.visible) {
+      void this.panel.webview.postMessage({ type: 'get' });
+    }
+  }
+
+  private getWebViewUri(...pathSegments: string[]) {
+    return this.panel?.webview.asWebviewUri(vscode.Uri.joinPath(this.context.getExtensionUri(), ...pathSegments));
+  }
+
+  private getHtmlForWebview(webview: vscode.Webview) {
```

Comment:
Just asking (don't know): does the functionality allow us to put it in a separate file?

Reply:
This is a very good point. It doesn't come out of the box using VS Code API but we should do it in future and extract to a separate file. 

This is out of scope of the current ticket because we have to consider performance penalty when serving HTML from disk and investigate a better architectural way of doing what we do now, perhaps by utilising a UI framework for bringing data into HTML.